app/javascript/components/pages/SearchEdit.js

```diff
@@ -1,13 +1,371 @@
-import React from "react"
-import PropTypes from "prop-types"
+import React, { Component } from "react";
+import { Button, Col, Form, FormGroup, Input, Label, Row } from "reactstrap";
+import { Redirect } from "react-router-dom";
+
+import PropTypes from "prop-types";
+import cityData from "../TicityCities.js";
+import findCities from "../../helpers/find-cities";
+
 class SearchEdit extends React.Component {
-    render() {
-        return (
-            <React.Fragment>
-                <h3>Edit Search Page</h3>
-            </React.Fragment>
-        );
-    }
+  constructor(props) {
+    super(props);
+    searchKey: "";
+    searchValue: 0;
+    this.state = {
+      form: {
+        ...this.props.search,
+        user_id: this.props.current_user_id,
+      },
+      success: false,
+    };
+  }
+
+  findCities = (cityData) => {
+    let { form } = this.state;
+    const newForm = findCities(
+      cityData,
+      form,
+      this.searchKey,
+      this.searchValue
+    );
+    this.setState({ form: newForm });
+  };
+
+  handleChange = (e) => {
+    let { form } = this.state;
+    this.searchKey = e.target.name;
+    this.searchValue = e.target.value;
+    form[e.target.name] = e.target.value;
+    this.setState({ form: form });
+  };
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.findCities(cityData);
+    this.props.updateSearch(this.state.form, this.props.search.id);
+    this.setState({ success: true });
+  };
+
+  render() {
+    return (
+      <>
+        <h3>Edit Categories</h3>
+        <div className="body-container">
+          <div className="form">
+            <Form>
+              <Row>
+                <Col>
+                  <FormGroup>
```

Comment:
This feels like it could be replaced with a loop. You could build an array of objects that have this shape

```
[{
  name: 'housing',
  label: 'Housing'
}]
```

Then loop through that array and avoid a lot of this copy-pasted code.

Reply:
we agree that this would be optimal, requested to icebox this for later once we reach full functionality